File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    answer = 0
    
    connected = [False] * n
    
    for k in range(n) :
        if connected[k] is False :
            deq = deque([k])
            answer += 1

            while len(deq) != 0 :
                v = deq.popleft()
                connected[v] = True
                for i in range(n) :
                    if connected[i] is False and computers[v][i] == 1:
                        deq.append(i)

    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.Queue;
import java.util.LinkedList;

class Solution {
    public int solution(int n, int[][] computers) {
        int len = computers.length;
        
        boolean[] vis = new boolean[len];
        
        int answer = 0;
        
        for(int i=0; i<len; i++) {
            if (!vis[i]) {
                answer += 1;
                vis[i] = true;
                Queue<Integer> q = new LinkedList<>();
                q.offer(i);
                while(q.size()!=0) {
                    int k = q.poll();
                    for (int j=0; j<len; j++) {
                        if (computers[k][j] == 1 && !vis[j]) {
                            vis[j] = true;
                            q.offer(j);
                        }
                    }
                }
            }
        }
        
        
        return answer;
    }
}